homeassistant/components/media_extractor/__init__.py

```diff
@@ -181,3 +180,19 @@ def get_stream_query_for_entity(self, entity_id: str | None) -> str:
             )
 
         return default_stream_query
+
+
+def get_best_stream(formats: list[dict[str, Any]]) -> str:
+    """Return the best quality stream.
+
+    As per
+    https://github.com/yt-dlp/yt-dlp/blob/master/yt_dlp/extractor/common.py#L128.
+    """
+
+    return cast(str, formats[len(formats) - 1]["url"])
+
+
+def get_best_stream_youtube(formats: list[dict[str, Any]]) -> str:
+    """YouTube requests also include manifest files. Skip them."""
+
+    return get_best_stream(list(filter(lambda format: "filesize" in format, formats)))
```

Comment:
let's use a list comprehension for readability: `[format for format in formats if "filesize" in format]`

Reply:
I tried to figure this out before but for some reason I couldn't get it working, but it turned out to be this simple, nice!